ConfigSpace/hyperparameters.pyx

```diff
@@ -806,23 +843,22 @@ cdef class UniformIntegerHyperparameter(IntegerHyperparameter):
         value = self._inverse_transform(value)
         return value
 
-    def _transform(self, vector: Union[np.ndarray, float, int]) -> Optional[Union[np.ndarray, float, int]]:
+    cpdef np.ndarray _transform_vector(self, np.ndarray vector):
+        if np.any(~np.isfinite(vector)):
```

Comment:
Hm, I just did some checks and think this can be sped up:
```
In [6]: %timeit np.any(np.isfinite(a))                                                                   
4.58 µs ± 45.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [9]: %timeit (~np.isfinite(a)).any()                                                                  
3.14 µs ± 46.6 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [4]: %timeit np.isnan(a).any()                                                                        
2.52 µs ± 21.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
Or like this:
```
In [11]: %timeit not np.isfinite(a).all()                                                             
2.73 µs ± 141 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
That's great! I think we can use `np.isnan(a).any()` as it is the fastest and you mentioned below that there are usually no infs/-infs around?